### ISSUE TITLE
Header-files in dist with mixed relative links

### ISSUE THREAD

**Issue by onkobu** [jcorporation/myMPD#242]
myMPD version: 6.2.3/ master

**Describe the bug**

When running make after checkout & cmake . (in-repo-build) some include-directives for relative header files in dist are wrong. There seems to be even a small set of files using both, ../dist and correct ../../dist. File list:

* src/mpd_client/mpd_client_api.c
* src/mpd_client/mpd_client_queue.c
* src/mpd_client/mpd_client_search.c
* src/mpd_client/mpd_client_state.c
* src/mpd_client/mpd_client_stats.c
* src/mpd_client/mpd_client_sticker.c
* src/mympd_api/mympd_api_bookmarks.c
* src/mympd_api/mympd_api_settings.c
* src/mympd_api/mympd_api_syscmds.c
* src/mympd_api/mympd_api_timer_handlers.c
* src/web_server/web_server_albumart.c
* src/web_server/web_server_utility.c

**To Reproduce**

1. checkout a take
2. cmake .
3. make
4. preprocessor errors complaining about non-existing header files in ../dist/...

Important to outline, that I did not use build.sh and also invoked cmake in repository's root instead of release-directory. After some build.sh clean* directory release is wiped clean and CMakeCache.txt in repository root is left behind, blocking all further attempts to build in release directory. A CMakeCache.txt in repository root directory needs to be cleaned, too.

The error from above does not occur when starting with build.sh on a clean repo (which wasn't obvious to me skipping some releases).

**Expected behavior**

Builds right away/ header files referenced through ../../dist or notifies about in-tree-builds (I come across this more and more often, ``cmake ..`` from a build-directory in tree refuses to build until the directory is outside of the tree. Must be a setting of cmake/ cmake-config of project.)

**Server plattform (please complete the following information):**
 - OS: Gentoo
 - OS version -
 - Plattform: x86_64
 - MPD version: -
 - Libmpdclient Version: -

**Debug logs (please attach if it can be usefull)**
```
/home/onkobu/git/myMPD/src/web_server/web_server_utility.c:7:10: fatal error: ../dist/src/sds/sds.h: No such file or directory
    7 | #include "../dist/src/sds/sds.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

**Comment by jcorporation** [comment]
Thanks for this bug report, wil be fixed with 6.3.0

**Comment by onkobu** [comment]
Thanks in advance.
